linker/shared/src/test/scala/org/scalajs/linker/frontend/optimizer/IntegerDivisionsTest.scala

```diff
@@ -0,0 +1,639 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.frontend.optimizer
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.ir.Names._
+import org.scalajs.ir.Trees._
+import org.scalajs.ir.Types._
+
+import org.scalajs.linker.testutils.TestIRBuilder._
+import org.scalajs.linker.testutils.TreeDSL._
+
+import IntegerDivisions._
+
+class IntegerDivisionsTest {
+  // We use many binary ops in these tests
+  import BinaryOp._
+
+  private val inum = VarRef(NumeratorArgName)(IntType)
+  private val lnum = VarRef(NumeratorArgName)(LongType)
+
+  /* For tests, we always use the configuration with WebAssembly and without
+   * RuntimeLong. This configuration produces code that is easier to match
+   * against and to interpret.
+   *
+   * Note that we test the code generated for divisions that the optimizer
+   * would choose *not* to actually use with WebAssembly (namely, long
+   * divisions by non-powers of 2), so we are not missing coverage compared to
+   * what JavaScript does.
+   *
+   * For the details of the JavaScript-specific generated code, we rely on the
+   * test suite.
+   */
+  val integerDivisions =
+    new IntegerDivisions(useRuntimeLong = false, isWebAssembly = true)
+
+  // ----------------------------------------------------
+  // --- Unit tests for the computation of magic data ---
+  // ----------------------------------------------------
+
+  @Test def testComputeSignedMagicInt(): Unit = {
+    def test(expectedM: Int, expectedAdd: Int, expectedShift: Int, divisor: Int): Unit = {
+      assertEquals(
+          divisor.toString(),
+          MagicData(expectedM, expectedAdd, expectedShift),
+          computeSignedMagic(Math.abs(divisor), divisor < 0))
+    }
+
+    // From section 10-5
+
+    test(0x6db6db6d, -1, 2, -7)
+
+    // Test cases from Hacker's Delight, 10-14, Table 10-1, Signed column
+
+    test(0x99999999, 0, 1, -5)
+    test(0x55555555, -1, 1, -3)
+    test(0x55555556, 0, 0, 3)
+    test(0x66666667, 0, 1, 5)
+    test(0x2aaaaaab, 0, 0, 6)
+    test(0x92492493, 1, 2, 7)
+    test(0x38e38e39, 0, 1, 9)
+    test(0x66666667, 0, 2, 10)
+    test(0x2e8ba2e9, 0, 1, 11)
+    test(0x2aaaaaab, 0, 1, 12)
+    test(0x51eb851f, 0, 3, 25)
+    test(0x10624dd3, 0, 3, 125)
+    test(0x68db8bad, 0, 8, 625)
+
+    // For completeness, but we never use the results of this function for powers of 2
+    for (k <- 1 to 30) {
+      test(0x7fffffff, -1, k - 1, -(1 << k))
+      test(0x80000001, 1, k - 1, 1 << k)
+    }
+  }
+
+  @Test def testComputeUnsignedMagicInt(): Unit = {
+    def test(expectedM: Int, expectedAdd: Int, expectedShift: Int, divisor: Int): Unit = {
```

Comment:
```suggestion
    def test(divisor: Int, expectedM: Int, expectedAdd: Int, expectedShift: Int): Unit = {
```

So the table is easier to compare with the book.